### PR TITLE
feat: add columns comments to iceberg

### PR DESCRIPTION
**Pull request by frenchytheasian**
### Feature or Bugfix
- Feature

### Detail
- Add ability to add columns comments to iceberg table using the wr.athena.to_iceberg function.

### Relates
- [https://github.com/aws/aws-sdk-pandas/issues/2469](https://github.com/aws/aws-sdk-pandas/issues/2469)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: d2252440345c8d3560c9c8fa1befdb5f40e1288b
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3cb30013-a98c-426f-a62c-fb23fd004672%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 9e948269089b2775dfbfbd3b7cb71386ceb7bf8b
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9b7271a1-a961-4bc8-be17-f7e31daf0ae5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: d2252440345c8d3560c9c8fa1befdb5f40e1288b
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3e34ec16-d97d-4c1b-8da6-d1fafeda7e39%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 9e948269089b2775dfbfbd3b7cb71386ceb7bf8b
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7163f195-65d5-4afe-8b6b-69ebf63f97fd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: d564c2f8112cfd10a5c98983dd5fa8388ff66e79
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=93bbb81a-2618-4fbb-94ea-8f6f2f03d4c4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: d564c2f8112cfd10a5c98983dd5fa8388ff66e79
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a6582def-08d3-4d53-9f5c-9839cf0d7fb9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 494c5fe28ae20f649ef6228aca42002c08a471e6
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c67c62d1-fe06-43cb-9371-195b7cf8874c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 494c5fe28ae20f649ef6228aca42002c08a471e6
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fd0672d7-3fe0-4897-a677-0225e3c51b1f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/athena/_write_iceberg.py [hunk 1]
```diff
@@ -196,6 +205,7 @@ def to_iceberg(
     dtype: Optional[Dict[str, str]] = None,
     catalog_id: Optional[str] = None,
     schema_evolution: bool = False,
+    columns_comments: Optional[Dict[str, str]] = None,
```
Comment (LeonLuttenberger) [inline awswrangler/athena/_write_iceberg.py:208]: I think this parameter should be wrapped inside of `glue_table_settings`, like it is [here](https://github.com/aws/aws-sdk-pandas/blob/main/awswrangler/s3/_write_parquet.py#L337).

File: awswrangler/athena/_write_iceberg.py [hunk 2]
```diff
@@ -294,6 +306,10 @@ def to_iceberg(
             "Either path or workgroup path must be specified to store the temporary results."
         )
 
+    glue_table_settings: Union[GlueTableSettings, None] = None
+    if columns_comments is not None:
+        glue_table_settings = GlueTableSettings(columns_comments=columns_comments)
+
```
Comment (LeonLuttenberger) [inline awswrangler/athena/_write_iceberg.py:312]: If the argument is changed to be `glue_table_settings`, you can get column comments out of it like this:
```python
glue_table_settings = cast(
    GlueTableSettings,
    glue_table_settings if glue_table_settings else {},
)

columns_comments = glue_table_settings.get("columns_comments")
```

File: awswrangler/athena/_write_iceberg.py [hunk 3]
```diff
@@ -35,12 +36,20 @@ def _create_iceberg_table(
...
 ) -> None:
     if not path:
         raise exceptions.InvalidArgumentValue("Must specify table location to create the table.")
 
     columns_types, _ = catalog.extract_athena_types(df=df, index=index, dtype=dtype)
-    cols_str: str = ", ".join([f"{k} {v}" for k, v in columns_types.items()])
+    cols_str: str = ", ".join(
+        [
+            f"{k} {v}"
+            if (columns_comments is None or columns_comments[k] is None)
```
Comment (LeonLuttenberger) [inline awswrangler/athena/_write_iceberg.py:48]: `column_comments[k]` will fail if `column_comments` is a dict but just doesn't have an entry for `k`. You can fix this by changing this line into:
```python
if (columns_comments is None or columns_comment.get(k) is None)
```

**Review by LeonLuttenberger — COMMENTED** [review]
Thanks for adding this. Looks good overall!

I'd also suggest adding a unit test to `tests/unit/test_athena.py` which checks if column comments are applied. The test would look something like this:
```python
def test_athena_to_iceberg_column_comments(
    path: str, path2: str, glue_database: str, glue_table: str
) -> None:
    df = pd.DataFrame({"c0": [0, 1, 2], "c1": [3, 4, 5]})
    column_comments = {
        "c0": "comment 0",
        "c1": "comment 1",
    }
    wr.athena.to_iceberg(
        df=df,
        database=glue_database,
        table=glue_table,
        table_location=path,
        temp_path=path2,
        keep_files=False,
        glue_table_settings={
            "columns_comments": column_comments,
        },
    )

    column_comments_actual = wr.catalog.get_columns_comments(glue_database, glue_table)

    assert column_comments_actual == column_comments
```

**Comment by frenchytheasian** [comment]
@LeonLuttenberger Thanks for the review! I will get these changes added in. I noticed that the majority of the tests weren't done using moto. Is there a reasoning behind this? If it makes sense, I would be interested in trying to write this test using moto and porting some of the existing tests over to use moto as well (in future PRs). I personally would prefer this to avoid costs in my AWS account, but if there's reasoning for not using moto, I am also fine with writing the test as normal.

**Comment by LeonLuttenberger** [comment]
The main reason is that Moto just didn't have support for all the services we needed to test. Some of our API functions make use of a wide range of services (e.g. `to_parquet` covers S3, Athena, Glue, KMS, Lake Formation), so the most reliable way to test them was to just do it on AWS directly.

For this specific test, what we want to actually test is that the Athena SQL query is indeed creating the Iceberg table with the right column comments. I don't think Moto would support anything like this.

I do agree though that it's worth looking into which unit test can be done just using Moto. The last time we did this exercise was a a few years ago, and I'm sure Moto has been improved in the meantime. However, I would keep it outside the scope of this PR.

I completely understand wanting to avoid all the costs in your AWS account. I'm happy to add that test myself, so that you don't have to incur those costs.

**Comment by frenchytheasian** [comment]
That makes complete sense. Thanks for clearing that up! I'm fine with adding the tests and these changes, but probably will not be able to get to it until middle of next week. If you want to get this PR in sooner, then feel free to add in the changes you suggested.

**Review by jaidisido — APPROVED** [review]
Approved once changes requested by @LeonLuttenberger are in. Thanks!

**Comment by frenchytheasian** [comment]
I made changes based on @LeonLuttenberger suggestions. One thing I realized is that column comments would only be successfully added if this function is creating a new table. I noted it in the documentation, but wanted to ask if that is something that I should throw into this PR or if I should open in another PR?

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: e322cc250663ebcac4ab6dabb07abfd4116dab0b
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c1669d35-20a6-4748-8137-390c348f27b7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: e322cc250663ebcac4ab6dabb07abfd4116dab0b
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f4de40e8-1f9b-4519-a31f-338e68330d35%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 928e45a41df8f7edce1b3f221d071cdf1668ce5c
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=05e6c169-209d-4d6c-b1ef-045a86d4218e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 31726cb15c18d8f78002df2a32000d21ab1c1717
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1e7ee93f-f7f8-48b6-b805-fc345cd58bf7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 928e45a41df8f7edce1b3f221d071cdf1668ce5c
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3cbc256e-e358-42ff-9ac8-57aa4b13904b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 31726cb15c18d8f78002df2a32000d21ab1c1717
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b0966eb5-052f-4b8f-bda6-ddf935df246e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by LeonLuttenberger** [comment]
> I made changes based on @LeonLuttenberger suggestions. One thing I realized is that column comments would only be successfully added if this function is creating a new table. I noted it in the documentation, but wanted to ask if that is something that I should throw into this PR or if I should open in another PR?

I think it makes sense to do it as a separate PR. Thanks for noting it in the documentation!

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 086015be5b9bd5260497c3e800f19d28fbf9321b
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ce6b062b-1406-4e0c-800a-05c9e312cef8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 086015be5b9bd5260497c3e800f19d28fbf9321b
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bf691b42-97ce-4250-a8d8-6b72bb52b3c8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**